### ISSUE TITLE
Segmentation Fault with duplicate arguments to dladm

### ISSUE THREAD

**Issue by colstrom** [joyent/illumos-joyent#141]
```
[root@gz] dladm show-link -o link,class,mtu,state,bridge,over,link
dladm: warning: Not enough space
Segmentation Fault (core dumped)
```

Removing the second `link` from the arguments proceeds without error.

Tested on `joyent_20170315T210557Z`

**Comment by ikozhukhov** [comment]
confirm it:

root@srv171:/var/cores# pstack core.dladm.808
core 'core.dladm.808' of 808:   dladm show-link -o link,class,mtu,state,bridge,over,link
 fffffd7ffe3458c9 ofmt_print (0, fffffd7fffdf90f0) + 19
 000000000040e5f2 show_link (4c5fb8, 1, fffffd7fffdffaf0) + 62
 fffffd7ffe3d0078 dladm_walk_datalink_id (40e590, 4c5fb8, fffffd7fffdffaf0, 3ff, 100000000, 1) + b8
 000000000040f72e do_show_link (3, fffffd7fffdffc40, 41c820) + 2de
 000000000041c75c main (4, fffffd7fffdffc38) + ac
 000000000040bad3 _start_crt () + 83
 000000000040ba38 _start () + 18
root@srv171:/var/cores# mdb core.dladm.808
Loading modules: [ libumem.so.1 libcmdutils.so.1 libc.so.1 libnvpair.so.1 ld.so.1 ]
> $C
fffffd7fffdf90d0 libinetutil.so.1`ofmt_print+0x19(0, fffffd7fffdf90f0)
fffffd7fffdff9a0 show_link+0x62(4c5fb8, 1, fffffd7fffdffaf0)
fffffd7fffdffa80 libdladm.so.1`dladm_walk_datalink_id+0xb8(40e590, 4c5fb8, fffffd7fffdffaf0, 3ff, 
100000000, 1)
fffffd7fffdffba0 do_show_link+0x2de(3, fffffd7fffdffc40, 41c820)
fffffd7fffdffbe0 main+0xac(4, fffffd7fffdffc38)
fffffd7fffdffc10 _start_crt+0x83()
fffffd7fffdffc20 _start+0x18()

**Comment by rzezeski** [comment]
I've filed [OS-6057](https://smartos.org/bugview/OS-6057) and added this to my queue. Thanks for the report.

**Comment by ikozhukhov** [comment]
additional example:

root@srv171:/var/cores# dladm show-link -o class,mtu,state,bridge,over,link
CLASS     MTU    STATE    BRIDGE     OVER                          LINK
phys      9000   up       --         --                            igb0
phys      1500   unknown  --         --                            igb1
phys      9000   unknown  --         --                            igb2
phys      1500   unknown  --         --                            igb3
vnic      9000   up       --         igb0                          kvm1
vnic      9000   up       --         igb0                          kvm2
vnic      9000   up       --         igb0                          kvm3
vnic      9000   up       --         igb0                          kvm4
vnic      9000   up       --         igb0                          vz174
vnic      9000   up       --         igb0                          vz173
vnic      9000   up       --         igb0                          vz172
vnic      9000   up       --         igb0                          vz170
root@srv171:/var/cores# dladm show-link -o class,mtu,state,bridge,over,link,link
dladm: warning: Not enough space
Segmentation Fault (core dumped)

as you can see - we are failed if we try to add 2 link field to print.

**Comment by ikozhukhov** [comment]
it is problem not in duplicate 'link' field - it is issue with long print - more 80 characters in line:
https://cl.ly/0T1M3t2W0E3E

**Comment by rzezeski** [comment]
This was fixed in c5bf6dbcaf952d0108897703c878c1c0eb4be27f.
